src/buffer.rs

```diff
@@ -395,7 +395,16 @@ impl<'p, D: SurfaceMemoryDescriptor, T: Borrow<Surface<D>>> CodedBuffer<'p, D, T
     pub fn new(picture: &'p Picture<PictureSync, T>) -> Result<Self, VaError> {
         let mut addr = std::ptr::null_mut();
         let mut segments = Vec::new();
-        let buffer = picture.coded_buffer().unwrap();
+
+        let buffer = match picture.coded_buffer() {
+            Some(buffer) => buffer,
+            None => {
+                return Err(bindings::constants::VA_STATUS_ERROR_INVALID_PARAMETER
```

Comment:
I thought about doing this too, but the thing is, if I see VA_STATUS_ERROR_* I will immediately try to debug the driver, because these errors are defined and returned by the driver itself.

If we start returning our own VA-API errors in cros-libva it can get confusing IMHO.

For example, someone running the VA tracer will see `VA_STATUS_SUCCESS` and yet may be greeted with `VA_STATUS_ERROR_INVALID_PARAMETER`.

Reply:
I can introduce a new `thiserror::Error` for this, but if we use the approach mention in [comment](https://github.com/chromeos/cros-libva/pull/3#issuecomment-1691247891) we could eliminate the need for this. WDYT?

Reply:
@bgrzesik I am not against what you proposed, but let's see whether Alex comes up with another idea, as he said he was going to think about it some more